src/components/pages/Authentication/hooks.ts

```diff
@@ -5,32 +5,61 @@ import { useLocation, useNavigate } from "react-router-dom";
 import { useAuth, useCleanUrl } from "@reearth/auth";
 import { useGetTeamsQuery } from "@reearth/gql";
 import { useT } from "@reearth/i18n";
-import { useTeam, useNotification } from "@reearth/state";
+import { useTeam, useNotification, useUserId } from "@reearth/state";
 
 // TODO: move hooks to molecules (page components should be thin)
 export default () => {
   const { isAuthenticated, isLoading, error: authError, logout, login } = useAuth();
   const [error] = useCleanUrl();
   const navigate = useNavigate();
   const location = useLocation();
-  const [currentTeam, setTeam] = useTeam();
+  const [currentWorkspace, setCurrentWorkspace] = useTeam();
+  const [currentUserId, setCurrentUserId] = useUserId();
   const [, setNotification] = useNotification();
   const passwordPolicy = window.REEARTH_CONFIG?.passwordPolicy;
   const t = useT();
 
   const { data, loading } = useGetTeamsQuery({ skip: !isAuthenticated });
-  const teamId = currentTeam?.id || data?.me?.myTeam.id;
+
+  if (isAuthenticated && !currentUserId) {
+    setCurrentUserId(data?.me?.id);
+  }
+  const workspaceId = currentWorkspace?.id || data?.me?.myTeam.id;
 
   useEffect(() => {
     if (location.pathname === "/login" && !new URLSearchParams(window.location.search).has("id"))
       login();
   }, [login, location.pathname]);
 
+  const handleRedirect = useCallback(
+    (workspaceId: string) => {
+      if (currentUserId == data?.me?.id) {
+        setCurrentWorkspace(
+          workspaceId
+            ? data?.me?.teams.find(t => t.id === workspaceId) ?? data?.me?.myTeam
+            : undefined,
+        );
+        navigate(`/dashboard/${workspaceId}`);
+      } else {
+        setCurrentUserId(data?.me?.id);
+        setCurrentWorkspace(data?.me?.myTeam);
+        navigate(`/dashboard/${data?.me?.myTeam.id}`);
+      }
+    },
+    [
+      currentUserId,
+      data?.me?.id,
+      data?.me?.teams,
+      data?.me?.myTeam,
+      setCurrentWorkspace,
+      navigate,
+      setCurrentUserId,
+    ],
+  );
```

Comment:
Is there a difference you feel between the values such as data, currentUserId, etc and then workspaceId? Why i ask is because you are passing workspaceId as a prop, but the other values you use directly and add as deps.

Reply:
yes you are right , I handle it same as others

Reply:
48b47ad76a7e35bbd1b45814258e6e94781b712f